File: themes/lib/targets/neovide.py
"""Neovide GUI font configuration (standalone Lua)."""

from themes.lib.schema import ColorScheme, ThemeState

TARGET_NAME = "neovide"
ASSEMBLY = "standalone"
OUTPUT_PATH = "~/.config/nvim/lua/neovide-theme.lua"
RELOAD_CMD = None
COMMENT = "--"


def generate(colors: ColorScheme, state: ThemeState) -> str:
    font = state.mono_font
    size = state.mono_font_size_for(TARGET_NAME)
    return f'vim.o.guifont = "{font}:h{size}"\n'

File: desktopctl/src/theme/targets/neovide.rs
use super::{Assembly, GeneratedContent, TargetMetadata};
use crate::theme::schema::{ColorScheme, ThemeState};

pub const METADATA: TargetMetadata = TargetMetadata::new(
    "neovide",
    Assembly::Standalone,
    &[
        "mono_font",
        "mono_font_size",
        "neovide_mono_font_size_offset",
    ],
)
.output("~/.config/nvim/lua/neovide-theme.lua")
.comment("--");

pub fn generate(_colors: &ColorScheme, state: &ThemeState) -> crate::Result<GeneratedContent> {
    Ok(GeneratedContent::text(format!(
        "vim.o.guifont = \"{}:h{}\"\n",
        state.mono_font,
        state.mono_font_size_for(METADATA.name)?,
    )))
}
